seqr/views/apis/anvil_workspace_api.py

```diff
@@ -193,29 +188,94 @@ def create_project_from_workspace(request, namespace, name):
+
+    return create_json_response({'projectGuid': project.guid})
+
+
+@anvil_auth_and_policies_required
+def add_workspace_data(request, project_guid):
+    """
+    Add data from an AnVIL workspace.
+
+    :param request: Django request object
+    :param project_guid: Django request object
+    :return success if no exceptions
+
+    """
+    project = get_project_and_check_permissions(project_guid, request.user, can_edit=True)
+    check_workspace_perm(request.user, CAN_EDIT, project.workspace_namespace, project.workspace_name, can_share=True)
+
+    request_json = json.loads(request.body)
+
+    missing_fields = [field for field in ['uploadedFileId', 'fullDataPath', 'vcfSamples'] if not request_json.get(field)]
+    if missing_fields:
+        error = 'Field(s) "{}" are required'.format(', '.join(missing_fields))
+        return create_json_response({'error': error}, status=400, reason=error)
+
+    pedigree_records = _parse_uploaded_pedigree(request_json, request.user)
+
+    previous_samples = Sample.objects.filter(
+        individual__family__project=project, is_active=True, elasticsearch_index__isnull=False,
+        dataset_type=Sample.DATASET_TYPE_VARIANT_CALLS).prefetch_related('individual')
+    previous_loaded_individuals = {s.individual.individual_id for s in previous_samples}
+    missing_loaded_samples = [individual_id for individual_id in previous_loaded_individuals if
+                              individual_id not in request_json['vcfSamples']]
+    if missing_loaded_samples:
+        return create_json_response({
+            'error': 'In order to add new data to this project, new samples must be joint called in a single VCF with all previously loaded samples.'
+                     ' The following samples were previously loaded in this project but are missing from the VCF: {}'.format(
+                ', '.join(sorted(missing_loaded_samples)))}, status=400)
+
+    _trigger_add_workspace_data(project, pedigree_records, request.user, request_json['fullDataPath'],
+                                previous_samples.first().sample_type,
+                                previous_loaded_ids=previous_loaded_individuals)
+
+    return create_json_response({'success': True})
+
+
+def _parse_uploaded_pedigree(request_json, user):
+    # Parse families/individuals in the uploaded pedigree file
+    json_records = load_uploaded_file(request_json['uploadedFileId'])
+    pedigree_records, _ = parse_pedigree_table(json_records, 'uploaded pedigree file', user=user, fail_on_warnings=True)
+
+    missing_samples = [record['individualId'] for record in pedigree_records
+                       if record['individualId'] not in request_json['vcfSamples']]
+
+    error = 'The following samples are included in the pedigree file but are missing from the VCF: {}'.format(
+                ', '.join(missing_samples)) if missing_samples else None
+
+    if error:
```

Comment:
it would be cleaner to do `if missing_samples` here and define the error message in the if block